### PR TITLE
Create idiomatic package structure

### PR DESCRIPTION
**Pull request by EnricoSchw**
These are some small changes:

- Separated main file from the lib folder
- Set up unit tests and github action

This PR is related to the closed PR: https://github.com/matrix-org/waterfall/pull/99

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.19
+      - name: Test
+        run: "./scripts/test.sh"
```
Comment (SimonBrandner) [inline .github/workflows/test.yaml:18]: Newline please

**Review by SimonBrandner — APPROVED** [review]
Otherwise LGTM

File: pkg/common/watchdog_test.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+)
+
+func BenchmarkWatchdogChannel_Notify(b *testing.B) {
+	watchdogConfig := common.WatchdogConfig{
+		Timeout:   2 * time.Second,
+		OnTimeout: func() {},
+	}
+	w := common.StartWatchdog(watchdogConfig)
+
+	// run the Fib function b.N times
```
Comment (daniel-abramov) [inline pkg/common/watchdog_test.go:17]: Sorry, what is the `Fib` function?

Comment (EnricoSchw) [reply pkg/common/watchdog_test.go:17]: Should mean Notify Function. I'll change this.

**Review by daniel-abramov — APPROVED** [review]

**Pull request merged**